contrib/src/test/java/com/datatorrent/contrib/cassandra/AbstractUpsertOutputOperatorCodecsTest.java

```diff
@@ -0,0 +1,488 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.lib.helper.OperatorContextTestHelper;
+import com.datatorrent.lib.helper.TestPortContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class AbstractUpsertOutputOperatorCodecsTest
+{
+
+  /***
+   * The schema that is used
+   *
+   *
+   *
+   *
+   * CREATE KEYSPACE unittests
+   * WITH replication = {
+   * 'class' : 'SimpleStrategy',
+   * 'replication_factor' : 1
+   * };
+   *
+   * CREATE TYPE unittests.address (
+   * street text,
+   * city text,
+   * zip_code int,
+   * phones set<text>
+   * );
+   *
+   * CREATE TYPE unittests.fullname ( firstname text, lastname text );
+   *
+   * CREATE TABLE unittests.users (
+   * userid text PRIMARY KEY,
+   * username FROZEN<fullname>,
+   * emails set<text>,
+   * top_scores list<int>,
+   * todo map<timestamp, text>,
+   * siblings tuple<int, text,text>,
+   * currentaddress FROZEN<address>,
+   * previousnames FROZEN<list<fullname>>
+   * );
+   *
+   * CREATE TABLE unittests.userupdates (
+   * userid text PRIMARY KEY,
+   * updatecount counter
+   * );
+   *
+   * CREATE TABLE unittests.userstatus (
+   * userid text,
+   * day int,
+   * month int,
+   * year int,
+   * employeeid text,
+   * currentstatus text,
+   * PRIMARY KEY ((userid,day,month,year), employeeid));
+   */
+
+  public static final String APP_ID = "TestCassandraUpsertOperator";
+  public static final int OPERATOR_ID_FOR_USER_UPSERTS = 0;
+
+
+  UserUpsertOperator userUpsertOperator = null;
+  OperatorContextTestHelper.TestIdOperatorContext contextForUserUpsertOperator;
+  TestPortContext testPortContextForUserUpserts;
+
+
+  @Before
+  public void setupApexContexts() throws Exception
+  {
+    Attribute.AttributeMap.DefaultAttributeMap attributeMap = new Attribute.AttributeMap.DefaultAttributeMap();
+    attributeMap.put(DAG.APPLICATION_ID, APP_ID);
+    contextForUserUpsertOperator = new OperatorContextTestHelper.TestIdOperatorContext(OPERATOR_ID_FOR_USER_UPSERTS,
+            attributeMap);
+    userUpsertOperator = new UserUpsertOperator();
+
+    Attribute.AttributeMap.DefaultAttributeMap portAttributes = new Attribute.AttributeMap.DefaultAttributeMap();
+    portAttributes.put(Context.PortContext.TUPLE_CLASS, User.class);
+    testPortContextForUserUpserts = new TestPortContext(portAttributes);
+
+    userUpsertOperator.setup(contextForUserUpsertOperator);
+    userUpsertOperator.activate(contextForUserUpsertOperator);
+    userUpsertOperator.input.setup(testPortContextForUserUpserts);
+  }
+
+  @Ignore
```

Comment:
can you give a quick try for using embedded Cassandra server for testing e.g. https://github.com/jsevellec/cassandra-unit/wiki/How-to-use-it-in-your-code